src/bp-core/bp-core-functions.php

```diff
@@ -5238,7 +5238,11 @@ function bp_core_profile_completion_steps_options() {
  *
  * @since BuddyBoss 1.4.9
  */
-function bp_core_xprofile_update_profile_completion_user_progress() {
+function bp_core_xprofile_update_profile_completion_user_progress( $user_id = '', $posted_field_ids = array(), $errors = array(), $old_values = array(), $new_values = array() ) {
+
+	if( empty( $user_id ) ){
+		$user_id 		= get_current_user_id();
```

Comment:
@milanhirapra Check line no 5247 [ we already assigning current user-id to $user_id  ], We have duplicate code. the solution is not correct.

Reply:
@dipskakadiya I forgot to remove this line but, I added the condition if the user id is null then assign the current user id.